app/controllers/courses/github_webhooks_controller.rb

```diff
@@ -0,0 +1,201 @@
+module Courses
+  class GithubWebhooksController < ApplicationController
+    include GithubWebhook::Processor
+    load_resource :course
+    skip_before_action :verify_authenticity_token
+    skip_before_action :authenticate_user!
+
+    def github_organization(payload)
+      case payload[:action]
+      when "member_invited"
+        student = @course.student_for_uid(payload[:invitation][:id])
+        return if student.nil?
+        student.org_membership_type = "Invited"
+        student.save
+      when "member_added"
+        student = @course.student_for_uid(payload[:membership][:user][:id])
+        return if student.nil?
+        student.is_org_member = true
+        student.org_membership_type = payload[:membership][:role].capitalize
+        student.save
+      when "member_removed"
+        student = @course.student_for_uid(payload[:membership][:user][:id])
+        return if student.nil?
+        student.is_org_member = false
+        student.org_membership_type = nil
+        student.save
+      when "renamed"
+        @course.course_organization = payload[:organization][:login]
+        @course.save
+      else
+        return
+      end
+    end
+
+    def github_repository(payload)
+      repo_info = payload[:repository]
+      existing_repo = GithubRepo.where(course: @course, repo_id: repo_info[:id]).first
+      case payload[:action]
+      when "created"
+        existing_repo.try(:destroy)
+        repo_visibility = repo_info[:private] ? "private" : "public"
+        GithubRepo.create(course: @course, name: repo_info[:name], full_name: repo_info[:full_name], url: repo_info[:html_url], repo_id: repo_info[:id], visibility: repo_visibility)
```

Comment:
Add repository contributor for initial creator based on sender OR just make an api call to the repo